pkg/operator/ceph/cluster/mgr/spec_test.go

```diff
@@ -93,6 +93,12 @@ func TestPodSpec(t *testing.T) {
 		assert.Equal(t, "CEPH_ARGS", d.Spec.Template.Spec.Containers[1].Env[len(d.Spec.Template.Spec.Containers[1].Env)-1].Name)                                                          // connection info to the cluster
 		assert.Equal(t, "-m $(ROOK_CEPH_MON_HOST) -k /etc/ceph/admin-keyring-store/keyring", d.Spec.Template.Spec.Containers[1].Env[len(d.Spec.Template.Spec.Containers[1].Env)-1].Value) // connection info to the cluster
 	})
+
+	t.Run(("check mgr ConfigureProbe"), func(t *testing.T) {
+		container := c.makeMgrDaemonContainer(&mgrTestConfig)
+		assert.NotNil(t, container.LivenessProbe)
+		assert.NotNil(t, container.StartupProbe)
+	})
```

Comment:
This is the kind of test we need to check that liveness/startup probes are set on pods. :+1:

We should also verify that the probe override logic is being used by the code that creates pods/deployments. The current test only verifies that probes exist but does not verify correctness of the implementation.

To verify correctness, in addition to this check we should change some configuration of the probes during the test setup (e.g., set a ridiculously large value for one field of each probe like 9999 seconds). The values should be different for each probe (e.g., one is 9999 seconds, one is 8888 seconds). Then, at this section of the test, we want to read back those values and ensure they are as we expect them. This will verify several things:
- that the probe override code path is used in the function
- that the liveness probe config propagates to the liveness probe only (i.e., the startup probe config does not override the liveness probe)
- that the startup probe config propagates to the startup probe only (i.e., the liveness probe config does not override the startup probe)
- that the value used for the probes comes from the right daemon config (i.e., the mgr's probe does not take its configuration from the mon or osd probe configs)

Finally, we want to implement the same type of test for mon, osd, mds, and rgw.

Reply:
Done, thanks for telling the deep instructions.